CI tests are failing

- Failure at https://github.com/computationalmodelling/nbval/runs/8236634175?check_suite_focus=true#step:5:80

- became known while running CI for merge request #184, which - however - only changes the Readme of the project (https://github.com/computationalmodelling/nbval/pull/184/files).

Speculation regarding the (first) failure (https://github.com/computationalmodelling/nbval/runs/8236634175?check_suite_focus=true#step:5:80):
```
=================================== FAILURES ===================================
_____________________ tests/sample_notebook.ipynb::Cell 9 ______________________
Notebook cell execution failed
Cell 9: Cell outputs differ

Input:
%matplotlib inline
import matplotlib.pyplot as plt
plt.plot([1, 2, 3, 4, 5], [1, 3, 2, 2.5, 1.5], 'o-', label='graph')
plt.xlabel('x')
plt.ylabel('y')
plt.title('Simple plot')
plt.legend(loc='best')

Traceback:
 mismatch 'text/plain'

 assert reference_output == test_output failed:

  '<Figure size... with 1 Axes>' == '<Figure size... with 1 Axes>'
  - <Figure size 640x480 with 1 Axes>
  ?              - ^ ^ ^
  + <Figure size 432x288 with 1 Axes>
  ?               ^^ ^ ^
```

Perhaps the default figure size changed in a new version of matplotlib / jupyter / ... ? I

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -10,7 +10,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.6, 3.7, 3.8, 3.9]
+        python-version: [3.7, 3.8, 3.9, '3.10']
 
     steps:
     - name: Checkout

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -24,8 +24,8 @@ jobs:
     - name: Install dependencies
       run: |
         python3 -m pip install --upgrade pip doit
-        python3 -m pip install .
         doit install_test_deps
+        python3 -m pip install .
         python3 -m pip freeze  # do make debugging easier for failures due to version changes
 
     - name: Test with pytest

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -26,6 +26,7 @@ jobs:
         python3 -m pip install --upgrade pip doit
         python3 -m pip install .
         doit install_test_deps
+        python3 -m pip freeze  # do make debugging easier for failures due to version changes
 
     - name: Test with pytest
       run: |

--- a/dodo.py
+++ b/dodo.py
@@ -27,7 +27,9 @@ def task_test():
     }
 
 def task_install_test_deps():
-    test_deps = ['matplotlib', 'sympy', 'pytest-cov', 'pytest-mock', 'nbdime']
+    # ipython_genutils is an indirect dependency of nbdime, but can be removed from this list
+    # once https://github.com/jupyter/nbdime/pull/618 ends up in a release
+    test_deps = ['matplotlib', 'sympy', 'pytest-cov', 'pytest-mock', 'nbdime', 'ipython_genutils']
     return {
         'actions': [_make_cmd(['pip', 'install'] + test_deps)],
     }

--- a/nbval/_version.py
+++ b/nbval/_version.py
@@ -1,2 +1,2 @@
-version_info = (0, 9, 6)
+version_info = (0, 9, '7.dev')
 __version__ = '.'.join(map(str, version_info[:3])) + ''.join(version_info[3:])

--- a/tests/sanitize_defaults.cfg
+++ b/tests/sanitize_defaults.cfg
@@ -14,3 +14,7 @@ replace: TIMESTAMP
 [Memory addresses]
 regex: (<[a-zA-Z_][0-9a-zA-Z_.]* at )(0x[0-9a-fA-F]+)(>)
 replace: \1MEMORY_ADDRESS\3
+
+[Matplotlib figure size]
+regex: (Figure size )\d+x\d+( with \d+ Axes)
+replace: \1WIDTHxHEIGHT\2

